server/src/main/java/org/elasticsearch/search/aggregations/bucket/BucketsAggregator.java

```diff
@@ -91,7 +105,8 @@ public final void collectBucket(LeafBucketCollector subCollector, int doc, long
      * Same as {@link #collectBucket(LeafBucketCollector, int, long)}, but doesn't check if the docCounts needs to be re-sized.
      */
     public final void collectExistingBucket(LeafBucketCollector subCollector, int doc, long bucketOrd) throws IOException {
-        if (docCounts.increment(bucketOrd, 1) == 1) {
+        int docCount = fieldDocCountProvider.getDocCount(doc);
+        if (docCounts.increment(bucketOrd, docCount) == 1) {
```

Comment:
Should it be changed to `docCounts.increment(bucketOrd, docCount) == docCount` ?

Reply:
Yes, it seems that I missed it. Thanks